### PR TITLE
[WIP][AIRFLOW-2910] Add HTTP URI support

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-2910

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

Setting up an HTTP connection is not intuitive.  This PR introduces support for a different connection URI format.
Related discusion: 
https://issues.apache.org/jira/browse/AIRFLOW-2910
CC: @ashb @potiuk 

When this solution is accepted, it will update the documentation.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Review by potiuk — CHANGES_REQUESTED** [review]
I think we need a few https:// test cases and possibly even backwards-compatibility test for url encoded http://https%...

File: airflow/models/connection.py [hunk 1]
```diff
@@ -128,6 +128,27 @@ def __init__(
...
+        if uri_parts.scheme in ['http', 'https']:
+            self._parse_from_http_uri(uri)
+        else:
+            self._parse_from_airflow_uri(uri)
+
+    def _parse_from_http_uri(self, uri):
+        self.conn_type = 'http'
+        url_parts = urlparse(uri)
+        netloc = "%s:%s" % (url_parts.hostname, url_parts.port) if url_parts.port else url_parts.hostname
+        # The fragment is used to store extra configuration. The HTTP client never uses fragments, so
```
Comment (ashb) [inline airflow/models/connection.py:140]: This needs documenting somewhere that end users will be able to see without finding this bit of the code.

File: tests/models/test_connection.py [hunk 2]
```diff
@@ -89,23 +89,37 @@ def test_connection_from_uri_without_extras(self):
...
         self.assertEqual(connection.port, 1234)
-        self.assertDictEqual(connection.extra_dejson, {'extra1': 'a value',
-                                                       'extra2': '/path/'})
+        self.assertDictEqual(connection.extra_dejson, {'extra1': 'a value', 'extra2': '/path/'})
+
+    def test_connection_from_http_uri_with_extras(self):
+        uri = 'http://user:password@host%2flocation:1234/schema#extra1=a%20value&extra2=%2fpath%2f'
+        connection = Connection(uri=uri)
+        self.assertEqual(connection.conn_type, 'http')
+        self.assertEqual(connection.host, 'http://host%2flocation:1234/schema')
```
Comment (ashb) [inline tests/models/test_connection.py]: This doesn't look like a "sensible" host value - is % even allowed in hostnames?

File: airflow/models/connection.py (hunk 1 above)
Comment (mik-laj) [reply airflow/models/connection.py:140]: Yes. Of course. First, I wanted to determine the implementations.

File: tests/models/test_connection.py (hunk 2 above)
Comment (mik-laj) [reply tests/models/test_connection.py]: I updated this PR. Currently, % is not allowed character in new format.

**Pull request closed without merging**